pkg/agent/sysadvisor/plugin/inference/modelresultfetcher/model_result_fetcher.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2022 The Katalyst Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package plugin is the package that defines the SysAdvisor plugin, and
+// those strategies must implement ModelResultFetcher interface.
+
+package modelresultfetcher
+
+import (
+	"context"
+	"sync"
+
+	"github.com/kubewharf/katalyst-core/pkg/agent/sysadvisor/metacache"
+	"github.com/kubewharf/katalyst-core/pkg/config"
+	"github.com/kubewharf/katalyst-core/pkg/metaserver"
+	metricspool "github.com/kubewharf/katalyst-core/pkg/metrics/metrics-pool"
+)
+
+type ModelResultFetcher interface {
+	FetchModelResult(ctx context.Context, metaReader metacache.MetaReader,
```

Comment:
`FetchModelResult` actually works as `UpdateInferenceResult`? should rename it to make it more clear?